packages/realm_dart/test/realm_logger_test.dart

```diff
@@ -1,211 +1,115 @@
+        });
+      }
+    }
+  });
 
-      RealmInternal.logMessageForTesting(RealmLogLevel.detail, "123");
+  test('Trace in subisolate seen in parent', () {
+    Realm.logger.setLogLevel(RealmLogLevel.off);
+    Realm.logger.setLogLevel(RealmLogLevel.all, category: RealmLogCategory.realm.sdk);
 
-      return await completer.future;
+    expectLater(Realm.logger.onRecord, emits(isA<RealmLogRecord>().having((r) => r.message, 'message', 'Hey')));
+    Isolate.run(() {
+      Realm.logger.log(RealmLogLevel.trace, 'Hey');
     });
-
-    expect(actual, LoggedMessage(RealmLogLevel.detail, "123"));
   });
 
-  test('Realm.logger supports logging from multiple isolates', () async {
-    List<LoggedMessage> actual = await Isolate.run(() async {
-      var completer = Completer<void>();
-      final result = <LoggedMessage>[];
-
-      Realm.logger.level = RealmLogLevel.all;
-      Realm.logger.onRecord.listen((event) {
-        if (event.message == "stop") {
-          return completer.complete();
-        }
-
-        result.add(LoggedMessage(event.level, event.message));
-      });
-
-      // run a second isolate to listen logging specific log level messages
-      await Isolate.run(() async {
-        var completer2 = Completer<void>();
-        Realm.logger.level = RealmLogLevel.error;
-        Realm.logger.onRecord.listen((event) {
-          if ((event.level == RealmLogLevel.error && event.message != "2") || (event.level == RealmLogLevel.trace && event.message != "3")) {
-            throw RealmError("Unexpected message ${LoggedMessage(event.level, event.message)}");
-          }
+  test('Trace in root isolate seen in subisolate', () async {
+    Realm.logger.setLogLevel(RealmLogLevel.off);
+    Realm.logger.setLogLevel(RealmLogLevel.all, category: RealmLogCategory.realm.sdk);
 
-          completer2.complete();
-        });
-
-        Future<void> logTestMessage(Level level, String message) async {
-          completer2 = Completer<void>();
-          RealmInternal.logMessageForTesting(level, message);
-          if (Realm.logger.level == RealmLogLevel.off) {
-            Future<void>.delayed(const Duration(milliseconds: 10)).then((value) {
-              completer2.complete();
-            }).ignore();
-          }
-          await completer2.future;
-        }
-
-        await logTestMessage(RealmLogLevel.error, "2");
-        await logTestMessage(RealmLogLevel.error, "2");
-
-        // turn off second isolate
-        Realm.logger.level = RealmLogLevel.off;
-
-        // log another message. second isoalte should not get it
-        await logTestMessage(RealmLogLevel.error, "first only");
+    final trace = Isolate.run(() async {
+      return (await Realm.logger.onRecord.first).message;
+    });
+    await Future<void>.delayed(const Duration(milliseconds: 100)); // yield
+    expectLater(trace, completion('Hey'));
+    Realm.logger.log(RealmLogLevel.trace, 'Hey');
```

Comment:
Do you mind sharing what the issue and solution was with this piece of code that we discussed earlier in the office?

Reply:
A native side bug in what became bd49967b.

Reply:
Great digging, might bug you about some of those changes at a later time once the release is out 👍